app/Repositories/BuildRepository.php

```diff
@@ -242,4 +249,53 @@ public function storeArtifact(
             ['disk' => config('filesystems.cloud')]
         );
     }
+
+    /**
+     * @param  Build  $build
+     * @param  string  $deviceId
+     * @return bool
+     */
+    public function isDeviceInRolloutRange(Build $build, string $deviceId): bool
+    {
+        // only devices that contacted the backend
+        // after this date are considered active
+        $threshold = now()
+            ->subDays(config('uppy.active_device_threshold'))
+            ->toDateTimeString();
```

Comment:
It's not clear to me the reason for this. What do these 30 days represent?

Reply:
It's to try to identify in some way the active users of an application: with this pr, we update every time the timestamp of the last time a device has done the check for updates. in this way, we can somehow identify the active devices of each application, in order to make the partial rollout a little more effective. If for example a device hasn't done the check for updates since 3 months, it means that the application hasn't been opened since 3 months, therefore uninstalled or unused, so I think it makes sense to exclude that device from the partial rollout range.